### ISSUE TITLE
Introduce concepts for locations and persons

### ISSUE THREAD

**Issue by kaikreuzer** [eclipse/smarthome#582]
migrated from Bugzilla [#443908](https://bugs.eclipse.org/bugs/show_bug.cgi?id=443908)
status NEW severity *enhancement* in component *Core* for *---*
Reported in version *unspecified* on platform *All*
Assigned to: Project Inbox

On 2014-09-12 04:45:51 -0400, Kai Kreuzer wrote:

> To better support use cases like geo-fencing, iBeacons / BLE tags, wearables and rules with user-context, we need to introduce new concepts for locations and persons.
> There were already discussions around this in the past, e.g. https://www.eclipse.org/forums/index.php/t/781501/ or https://bugs.eclipse.org/bugs/show_bug.cgi?id=423550

**Comment by pinkfish** [comment]
I don't think you need people, really, but locations are a very strong semantic link.  For doing occupancy in a useful way rooms/location are needed to be able to pull flags and stats up a level from things to the thing they are inside.  Making the rooms in a structure also gives the structure inherent information about occupancy too.  A movement sensor gives some level of occupancy but an alarm gives a much stronger signal, for example.  The alarm is at the structure level but a motion sensor is at the room level.  We could automatically track occupancy in locations/structures just by using motion sensor input in a room + alarms to control the top level state.

iBeacons and things like that help here too, but I think the gains you get from allowing general purpose automation to exist is pretty high.

**Comment by kaikreuzer** [comment]
Wow, this issue is already old...

I'd just like to revive it with a citation from a comment on https://github.com/openhab/openhab-android/pull/914#issuecomment-394777623:

_We could use Location items for persons and phones should then be assigned to (exactly) one of these items. If they were tagged with object:person, the app could do a simple rest request for all items with such a tag and ask the user to select one (this would need to be done somewhere in the main settings)._

_All our rooms/homes/offices/otherplaces are usually already modelled as Group items - if those were tagged with object:place (or whatever identifies an item to be a place), this list could be requested by the app and beacons and geofences could be assigned to those._

_To be able to do this, Location items should not only allow geo-coordinates as a state, but also place-item names, but this is a change that was anyhow considered when introducing Location items._

This concept would allow us to have minimal efforts wrt introducing new kinds of entities or concepts - we could rely on the semantic tags and still cover the major use cases.
